src/metabase/lib/util.cljc

```diff
@@ -48,8 +68,11 @@
           :stages   (inner-query->stages (:query query))}
          (dissoc query :type :query)))
 
-(mu/defn pipeline :- ::lib.schema/query
-  "Ensure that a `query` is a pMBQL `:pipeline` query."
+(mu/defn pipeline
```

Comment:
I think it makes more sense to have `pipeline` just do some of the basic shape transformations, and have `->pMBQL` be responsible for completely converting something to pMBQL, thus having the validation happen here doesn't make a ton of sense